### DISCUSSION TITLE
Please update your repository

### DISCUSSION THREAD

**Discussion by githubbapoopa** [YoshifumiFuyuno/Search-from-Popup-or-ContextMenu#52] [Q&A]
This is the single best addon I've ever used.

I tried to heavily promote it on the originators github

https://github.com/CanisLupus/swift-selection-search/issues/238

https://github.com/CanisLupus/swift-selection-search/issues/244

https://github.com/garywill/BigSearch/issues/6

the original developer agreed

https://github.com/CanisLupus/swift-selection-search/issues/244#issuecomment-1037404697

So thank you very much.

I'd like to ask you to update your repository so that your excellent work can be further improved.

Again Thank you!

**Comment by YoshifumiFuyuno** [comment]
Hi. Thank you for your compliment!:smiley:
I don't have time to write a comment for publication, and I'm planning to completely rewrite this add-on for MV3 support in the not too distant future, while maintaining the functionality as much as possible, so I'm not publishing it now. If MV3 becomes mandatory, the current code will be worthless.
I don't want to spend too much time on other things because I don't have much time before MV3 becomes mandatory.

For Chrome, June 2023 is the limit, even if you're using enterprise policy.
![Chrome Manifest V2 support timeline](https://wd.imgix.net/image/SHhb2PDKzXTggPGAYpv8JgR81pX2/zXdU3hdkj1K0Ks6tAfB4.png)
[Manifest V2 support timeline - Chrome Developers](
https://developer.chrome.com/docs/extensions/mv3/mv2-sunset/)
[Manifest v3 update | Mozilla Add-ons Community Blog](
https://blog.mozilla.org/addons/2021/05/27/manifest-v3-update/)

Best regards

**Comment by githubbapoopa** [comment]
Ok understood. Thanks that you have taken the time to explain your decision.

My opinion ist still: If the code i out, shit can happen and it doesn't matter. Like MV3 indefinetely delayed, Corona strikes, the sky falls down...

So please still consider to release your code in its actual form...

Others feel the same

https://github.com/CanisLupus/swift-selection-search/issues/244#issuecomment-1043498710

code is never worthless

Thanks!!!!

**Comment by cheater** [comment]
Hi @YoshifumiFuyuno,
It is understood that you would like to update the addon heavily sometime in the future. Thank you very much for all the work you have done and are doing on SPC. I agree with @gthbububba that this is a great piece of work. While this is happening, releasing the code as it is right now would be beneficial. Here are the reasons I think are important:

1. It's pretty important to be able to see the code that we install on our computers.
2. Releasing the code follows the spirit of open source. On the other hand, not releasing the code can be somewhat worrying to some people who are distrustful. You have released the plugin under the GPL license - but that would require releasing the code too.
3. For Firefox users who want SSS, but with some bug fixes and tune-ups, this is currently better than a fully rewritten version.
4. Personally I'd like to improve SPC, and add to it. I'd like to modify it for myself first, see what features work, and then let you know of new functionality, and write a pull request. And if the MV3 version comes out, I'd write the functionality on that. I wouldn't want to burden you with this while you're trying to do the MV3 version. One thing I'd like to work on is image search. But I'd also like to add some features that only I will use that I would not commit to the repository - I just want to customize.
5. If the old code has some edge case that the new code does not support, a person stuck with the old code will have reduced support.

Thanks!

**Reply by sphh** [reply]
I also would like to see the source code because of the points made by @cheater.

Then IMHO there are the licensing issues:

1. You say, that the software is published under GPL 3. This license requires that the source code is made available, ie. published.
2. The original [LICENSE](https://github.com/CanisLupus/swift-selection-search/blob/master/LICENSE) requires ‘The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.’ I could not find this copyright notice anywhere.

**Comment by githubbapoopa** [comment]
no pressure :)

**Comment by githubbapoopa** [comment]
https://github.com/garywill/BigSearch/issues/6

**Comment by cheater** [comment]
BigSearch is actually a good example. I looked through its code and it calls out and sends data to Chinese cloud providers. I don't like this because those companies are required by law to share their data with the Chinese government. Unfortunately, I can't look into the code of SPC!

**Reply by garywill** [reply]
BigSearch users have option to disable analystics
https://github.com/garywill/BigSearch/blob/92b79daaa59f238c2099c43ebc483c5993712ece/src/i_scripts.js#L1214-L1219
Even added some comments on the code to let people see how the analystics work
